### PR TITLE
Deprecated 3 field sort in Realm and RealmQuery

### PR DESCRIPTION
**Pull request by cmelchior**
Part of #1594. Merge all PR's in one release.

@realm/java 

Final commit message
----
Title: Deprecated 3 field sort methods

Having many overrides of the same method is a concept heavily used in e.g. Guava, however we need to be mindful about the method count and general surface of our public API. Also we are seeing no indication that these methods are being used. 

In this change we therefore deprecating 3 field sort and refer to the more generic multi-arg sort instead for those few that run into this use case.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 0.90.0
+
+### Deprecated
+
+* `RealmObject.allObjectsSorted(field, sort, field, sort, field, sort)`. Use `RealmObject.allObjectsSorted(field[], sort[])`` instead.
```
Comment (zaki50) [inline CHANGELOG.md]: `RealmObject.allObjectsSorted` should be `Realm.allObjectsSorted`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+* `RealmQuery.allObjectsSorted(field, sort, field, sort, field, sort)`. Use `RealmQuery.allObjectsSorted(field[], sort[])`` instead.
```
Comment (zaki50) [inline CHANGELOG.md]: `RealmQuery.allObjectsSorted` should be `RealmQuery.findAllSorted`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+* `RealmQuery.allObjectsSortedAsync(field, sort, field, sort, field, sort)`. Use `RealmQuery.allObjectsSortedAsync(field[], sort[])`` instead.
```
Comment (zaki50) [inline CHANGELOG.md]: `RealmQuery.allObjectsSortedAsync` should be `RealmQuery.findAllSortedAsync`

**Comment by zaki50** [comment]
👍  other than minor comments for `CHANGELOG.md`

**Comment by kneth** [comment]
:+1:

**Comment by cmelchior** [comment]
CI: Retest this

**Pull request merged**